internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+				Optional:    true,
+				Validators: []validator.String{
+					configvalidators.ValidUrl(),
+				},
+			},
+			"default_access_token_manager_ref": schema.SingleNestedAttribute{
+				Description: "The default access token manager for this client.",
+				Optional:    true,
+				Attributes:  resourcelink.ToSchema(),
+			},
+			"restrict_to_default_access_token_manager": schema.BoolAttribute{
+				Description: "Determines whether the client is restricted to using only its default access token manager. The default is false.",
+				Computed:    true,
+				Optional:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"validate_using_all_eligible_atms": schema.BoolAttribute{
+				Description: "Validates token using all eligible access token managers for the client. This setting is ignored if 'restrictToDefaultAccessTokenManager' is set to true.",
+				Computed:    true,
+				Optional:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"refresh_rolling": schema.StringAttribute{
+				Description: "Use ROLL or DONT_ROLL to override the Roll Refresh Token Values setting on the Authorization Server Settings. SERVER_DEFAULT will default to the Roll Refresh Token Values setting on the Authorization Server Setting screen. Defaults to SERVER_DEFAULT.",
+				Computed:    true,
+				Optional:    true,
+				Default:     stringdefault.StaticString("SERVER_DEFAULT"),
+				Validators: []validator.String{
+					stringvalidator.OneOf("ROLL", "DONT_ROLL", "SERVER_DEFAULT"),
+				},
+			},
+			"refresh_token_rolling_interval_type": schema.StringAttribute{
+				Description: "Use OVERRIDE_SERVER_DEFAULT to override the Refresh Token Rolling Interval value on the Authorization Server Settings. SERVER_DEFAULT will default to the Refresh Token Rolling Interval value on the Authorization Server Setting. Defaults to SERVER_DEFAULT.",
+				Computed:    true,
+				Optional:    true,
+				Default:     stringdefault.StaticString("SERVER_DEFAULT"),
+				Validators: []validator.String{
+					stringvalidator.OneOf("OVERRIDE_SERVER_DEFAULT", "SERVER_DEFAULT"),
+				},
+			},
+			"refresh_token_rolling_interval": schema.Int64Attribute{
+				Description: "The minimum interval to roll refresh tokens, in hours. This value will override the Refresh Token Rolling Interval Value on the Authorization Server Settings.",
+				Optional:    true,
+			},
+			"persistent_grant_expiration_type": schema.StringAttribute{
+				Description: "Allows an administrator to override the Persistent Grant Lifetime set globally for the OAuth AS. Defaults to SERVER_DEFAULT.",
+				Computed:    true,
+				Optional:    true,
+				Default:     stringdefault.StaticString("SERVER_DEFAULT"),
+			},
+			"persistent_grant_expiration_time": schema.Int64Attribute{
+				Description: "The persistent grant expiration time. -1 indicates an indefinite amount of time.",
+				Computed:    true,
+				Optional:    true,
+				Default:     int64default.StaticInt64(0),
```

Comment:
Should this default be -1? I assume 0 would make things immediately expire, which seems a little weird.

Reply:
If you don't send a value, this value is 0, by default